app/components/new-file/new-file.component.ts

```diff
@@ -21,58 +22,46 @@ export class NewFileResult {
 })
 export class NewFileComponent implements ModalComponent, AfterViewInit {
     set modalInfo(modalInfo: ModalInfo) {
-        let plugins: string[][] = modalInfo.data;
-        let availablePlugins: any = { subLists: {} };
+        const plugins: PluginData[] = modalInfo.data;
+        // This code is extremely evil but necessary.
```

Comment:
Couldn't this evil at least happen in the plugin service? Or better yet, why can't this just be done without the evil in core?